cpp/src/arrow/compute/api_scalar.h

```diff
@@ -1006,17 +1023,20 @@ ARROW_EXPORT Result<Datum> DayOfYear(const Datum& values, ExecContext* ctx = NUL
 ARROW_EXPORT
 Result<Datum> ISOYear(const Datum& values, ExecContext* ctx = NULLPTR);
 
-/// \brief ISOWeek returns ISO week of year number for each element of `values`.
+/// \brief Week returns week of year number for each element of `values`.
 /// First ISO week has the majority (4 or more) of its days in January.
-/// Week of the year starts with 1 and can run up to 53.
+/// Year can have 52 or 53 weeks. Week numbering can start with 0 or 1
+/// depending on DayOfWeekOptions.count_from_zero.
 ///
-/// \param[in] values input to extract ISO week of year from
+/// \param[in] values input to extract week of year from
+/// \param[in] options for setting numbering start
 /// \param[in] ctx the function execution context, optional
 /// \return the resulting datum
 ///
-/// \since 5.0.0
+/// \since 6.0.0
 /// \note API not yet finalized
-ARROW_EXPORT Result<Datum> ISOWeek(const Datum& values, ExecContext* ctx = NULLPTR);
+ARROW_EXPORT Result<Datum> Week(const Datum& values, WeekOptions options = WeekOptions(),
+                                ExecContext* ctx = NULLPTR);
```

Comment:
Also add a helper for USWeek?

Reply:
Done.

Reply:
It appears to be missing its counterpart in api_scalar.cc.

Reply:
Added.